File: airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/util/Flattening.java
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.integrations.destination.s3.util;

import com.fasterxml.jackson.annotation.JsonCreator;

public enum Flattening {

  NO("No flattening"),
  ROOT_LEVEL("Root level flattening");

  private final String value;

  Flattening(final String value) {
    this.value = value;
  }

  public String getValue() {
    return value;
  }

  @JsonCreator
  public static Flattening fromValue(final String value) {
    for (final Flattening f : Flattening.values()) {
      if (f.getValue().equalsIgnoreCase(value)) {
        return f;
      }
    }
    throw new IllegalArgumentException("Unexpected value: " + value);
  }

}

File: airbyte-cdk/java/airbyte-cdk/s3-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/s3/util/Flattening.kt
/*
 * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.cdk.integrations.destination.s3.util

import com.fasterxml.jackson.annotation.JsonCreator
import kotlin.jvm.Throws

enum class Flattening(val value: String) {
    NO("No flattening"),
    ROOT_LEVEL("Root level flattening");

    companion object {
        @JvmStatic
        @JsonCreator
        @Throws(IllegalArgumentException::class)
        fun fromValue(value: String): Flattening {
            for (f in entries) {
                if (f.value.equals(value, ignoreCase = true)) {
                    return f
                }
            }
            throw IllegalArgumentException("Unexpected value: $value")
        }
    }
}
